libs/@blockprotocol/core/src/service-handler.ts

```diff
@@ -10,9 +15,19 @@ import { GenericMessageCallback, MessageContents, MessageData } from "./types";
  */
 export abstract class ServiceHandler {
   /** the CoreHandler this service is registered with, for passing messages via */
-  readonly coreHandler: CoreHandler;
+  private _coreHandler: CoreHandler | null = null;
+
+  public get coreHandler(): CoreHandler | null {
+    return this._coreHandler;
+  }
```

Comment:
Why is this necessary? It looks like `coreHandler` is only used in this class, and could remain `private`

Reply:
It was public readonly before and I didn't want to make a breaking change if it wasn't necessary. Can remove if you think it being private is fine.